app/src/main/java/com/github/swent/echo/viewmodels/HomeScreenViewModel.kt

```diff
@@ -66,18 +66,24 @@ constructor(
             )
         )
     val filtersContainer = _filtersContainer.asStateFlow()
-
-    private val _profileName =
-        MutableStateFlow<String>("John Doe") // placeholder until we have user profiles
+    private val _profileName = MutableStateFlow<String>("")
     val profileName = _profileName.asStateFlow()
-    private val _profileClass =
-        MutableStateFlow<String>("IN - BA6") // placeholder until we have user profiles
+    private val _profileClass = MutableStateFlow<String>("")
     val profileClass = _profileClass.asStateFlow()
+    private var section = ""
+    private var semester = ""
 
     init {
         viewModelScope.launch {
+            val userId = authenticationService.getCurrentUserID() ?: ""
             allEventsList = SAMPLE_EVENTS // repository.getAllEvents()
-            allTagSet = SAMPLE_TAGS // repository.getAllTags()
+            allTagSet = SAMPLE_TAGS // repository.getAllTags().toSet()
```

Comment:
Why not use the repository to the all events and tags?

Reply:
lol my bad I debugged something and forgot to put it back